sdk/storage/azure-storage-queue/azure/storage/queue/_models.py

```diff
@@ -275,9 +277,20 @@ def __init__(self, command, results_per_page=None, continuation_token=None):
         )
         self._command = command
         self.results_per_page = results_per_page
+        self.messages_to_retrieve = messages_to_retrieve
 
     def _get_next_cb(self, continuation_token):
         try:
+            if self.messages_to_retrieve is not None:
```

Comment:
What if it is `None`? Would we ever expect it to be `None` since we control it? Probably safe to remove this check especially since we don't do anything if it is. If it somehow miraculously was `None` I'd think we'd want to let the exception happen so we stop the iteration.

Reply:
Maybe I'm misunderstanding so we can follow-up tomorrow; but I believe the reason why this check is valid is because we want to have the same codepath if the customer does not define a `max_messages` value (which would propagate to `messages_to_retrieve`), then we want `receieve_messages()` to work exactly like it did before the introduction of the keyword `max_messages`.

Reply:
Resolved in offline discussion.